File: frontend/src/types/lot.ts
import type { Animal } from './animal'

export interface Lot {
  id: string
  name: string
  field_id: string
}

export interface LotAnimalsResponse {
  animals: Animal[]
}

export interface AdgResponse {
  lot_id: string
  lot_name: string
  period: { from: string; to: string }
  animals_count: number
  avg_adg_kg_day: string | null
}

File: backend/app/schemas/lot.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class LotResponse(BaseModel):
    id: uuid.UUID
    name: str
    field_id: uuid.UUID

    model_config = {"from_attributes": True}


class LotAnimalsResponse(BaseModel):
    animals: list[dict]


class AdgResponse(BaseModel):
    lot_id: uuid.UUID
    lot_name: str
    period: dict
    animals_count: int
    avg_adg_kg_day: Decimal | None
